sphinx/builders/linkcheck.py

```diff
@@ -27,6 +28,19 @@
 from sphinx.util.http_date import rfc1123_to_epoch
 from sphinx.util.nodes import get_node_line
 
+
+class LinkStatus(StrEnum):
```

Comment:
Can we rename this to `_Status` or `_LinkStatus`, move the definition to after the `if TYPE_CHECKING` imports (then add a second `TYPE_CHECKING` block for the type aliases), and remove the `_Status` type aliase please? Other than that I think good to go.